### PR TITLE
[#169067698] fix loading potService in messageList

### PR DESCRIPTION
**Pull request by morrys**
this PR allows you to display, if present, the service previously loaded while it is being refreshed.

loading-service-message

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🐞 [#169067698](https://www.pivotaltracker.com/story/show/169067698): Quando si apre un messaggio in modalità offline e l'utente torna indietro, nella lista dei messaggi, il messaggio precedentemente aperto viene visualizzato in caricamento fino al timeout del servizio di  refresh

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/components/messages/MessageList.tsx [hunk 1]
```diff
@@ -227,6 +227,7 @@ class MessageList extends React.Component<Props, State> {
     const potService = this.props.servicesById[meta.sender_service_id];
     if (
       potService &&
+      !pot.isSome(potService) &&
```
Comment (Undermaken) [inline ts/components/messages/MessageList.tsx]: It works!
Anyaway we could rewrite it to improve logic and readability.
The condition to satisfied should be: we have to show a loading item if either service or message are in loading state.

```typescript
const potService = this.props.servicesById[meta.sender_service_id];
    const isServiceLoading = potService
      ? pot.isNone(potMessage) && pot.isLoading(potService) // is none loading
      : false;
    if (isServiceLoading || pot.isLoading(potMessage)) {
      return MessageListItemPlaceholder;
    }
```
Could it be better?

File: ts/components/messages/MessageList.tsx [hunk 2]
```diff
@@ -227,6 +227,7 @@ class MessageList extends React.Component<Props, State> {
     const potService = this.props.servicesById[meta.sender_service_id];
```
Comment (Undermaken) [inline ts/components/messages/MessageList.tsx]: It works!
Anyaway we could rewrite it to improve logic and readability.
The condition to satisfied should be: we have to show a loading item if either service or message are in loading state.

```typescript
const potService = this.props.servicesById[meta.sender_service_id];
    const isServiceLoading = potService
      ? pot.isNone(potMessage) && pot.isLoading(potService) // is none loading
      : false;
    if (isServiceLoading || pot.isLoading(potMessage)) {
      return MessageListItemPlaceholder;
    }
```
Could it be better?

Comment (morrys) [reply ts/components/messages/MessageList.tsx]: yes,
but this: 

> pot.isNone(potMessage) && pot.isLoading(potService) // is none loading 

it should be corrected like this: `pot.isNone(potService) && pot.isLoading(potService) // is none loading`

**Pull request merged**